FsSonarRunner/FsSonarRunner/FsSonarRunner.fsproj

```diff
@@ -1,124 +1,34 @@
-    </Reference>
-    <Reference Include="FSharp.Data">
-      <HintPath>..\packages\FSharp.Data.2.3.3\lib\net40\FSharp.Data.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="FSharpLint.Core">
-      <HintPath>..\packages\FSharpLint.Core.0.7.6\lib\net45\FSharpLint.Core.dll</HintPath>
-    </Reference>
-    <Reference Include="Microsoft.Build.Framework">
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="mscorlib" />
-    <Reference Include="FSharp.Core, Version=$(TargetFSharpCoreVersion), Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="nunit.framework">
-      <HintPath>..\packages\NUnit.3.7.1\lib\net45\nunit.framework.dll</HintPath>
-    </Reference>
-    <Reference Include="System" />
-    <Reference Include="System.Collections.Immutable">
-      <HintPath>..\packages\System.Collections.Immutable.1.3.1\lib\portable-net45+win8+wp8+wpa81\System.Collections.Immutable.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="System.Core" />
-    <Reference Include="System.Numerics" />
-    <Reference Include="System.Reflection.Metadata">
-      <HintPath>..\packages\System.Reflection.Metadata.1.4.2\lib\portable-net45+win8\System.Reflection.Metadata.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Xml" />
-    <Reference Include="System.Xml.Linq" />
-    <ProjectReference Include="..\FsSonarRunnerCore\FsSonarRunnerCore.fsproj">
-      <Name>FsSonarRunnerCore</Name>
-      <Project>{584f069c-db4c-4561-9402-5b9b703f15a8}</Project>
-      <Private>True</Private>
-    </ProjectReference>
+    <PackageReference Update="FSharp.Core" Version="4.6.2" />
+    <PackageReference Update="System.ValueTuple" Version="4.5.0" />
   </ItemGroup>
-  <Import Project="..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets" Condition="Exists('..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
-    <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
-    </PropertyGroup>
-    <Error Condition="!Exists('..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\FSharp.Compiler.Service.ProjectCracker.13.0.0\build\net45\FSharp.Compiler.Service.ProjectCracker.targets'))" />
+
+  <Target Name="PostBuild" AfterTargets="PostBuildEvent">

```

Comment:
arent these already run regardless if the project needs to compile, i have always been against pre/post build events.

Reply:
According to [Microsoft](https://docs.microsoft.com/en-us/cpp/build/how-to-use-build-events-in-msbuild-projects?view=vs-2019)  "the _post-build_ event occurs after the build successfully ends."

For sure, this can also be done in appveyor. My goal was to publish the currently supported rule set with each build, as I saw thatupdating FSharpLint changed it.